### PR TITLE
feat(ui): add a new UI component DropdownMenu (ENG-56423)

### PR DESCRIPTION
**Pull request by stephenle17**
`<DropdownMenu className="TestDropdownMenu" />`
image

```
const customValues = [
    {
        text: 'Go to new page',
        onClick: function () {
            window
                .open('https://localhost:6006/example_go_to_new_page', '_blank')
                .focus();
        },
    },
    {
        text: 'Do an action or popup',
        onClick: function () {
            window.alert('Do something');
        },
    },
];
```

image

`<DropdownMenu  className="TestDropdownMenu"  requestUrl="/-/custom_url"/>`
image

### CODE REVIEW

File: src/l10n/en.po [hunk 1]
```diff
@@ -56,6 +60,18 @@ msgstr ""
 msgid "Go back"
 msgstr ""
 
+#: 7bdc
```
Comment (echristie-bc) [inline src/l10n/en.po:63]: Why are these translation values commented out ?

File: src/l10n/en.po [hunk 2]
```diff
@@ -32,6 +32,10 @@ msgstr ""
 "Language-Team: \n"
 "Plural-Forms: \n"
 
+#: 0b0c
+msgid "{0}"
```
Comment (echristie-bc) [inline src/l10n/en.po:36]: Where is this translation value used?

Comment (stephenle17) [reply src/l10n/en.po:36]: That translation is caused by line 67
```
                            text: t`${item?.text}`,
```

It seems that I did not use the `{ t } from '@bluecateng/l10n.macro'` correctly, resulting in the language segment not being translated as intended.

I'll remove the translation on that line.

File: src/l10n/en.po (hunk 1 above)
Comment (stephenle17) [reply src/l10n/en.po:63]: In the initial design, the text was used directly, so when I used the command `npx @bluecateng/l10n-cli` for translation, the file was saved with the translated text. However, after the restructuring, the text was no longer used directly, and the command `npx @bluecateng/l10n-cli` these translation values commented out.
From: ``` t`Go to BAM` ```
To: ``` t`${item.text}` ```

**Review by echristie-bc — APPROVED** [review]

**Review by echristie-bc — APPROVED** [review]

**Pull request merged**